### PR TITLE
fix: node.js debugger adds stderr (but exit code is 0) -> shouldn't throw

### PR DESCRIPTION
**Pull request by FuPeiJiang**
https://github.com/nodejs/node-gyp/pull/2719#issuecomment-1266747060

___

Debugger attached.
Waiting for the debugger to disconnect...

### CODE REVIEW

**Review by cclauss — APPROVED** [review]

File: pylib/gyp/input.py [hunk 1]
```diff
@@ -961,33 +961,26 @@ def ExpandVariables(input, phase, variables, build_file):
...
-                        # Simulate check_call behavior, since check_call only exists
-                        # in python 2.5 and later.
+                    if result.returncode > 0:
                         raise GypError(
                             "Call to '%s' returned exit status %d while in %s."
-                            % (contents, p.returncode, build_file)
+                            % (contents, result.returncode, build_file)
                         )
-                    replacement = p_stdout.rstrip()
+                    replacement = result.stdout.decode("utf-8").rstrip()
```
Comment (gengjiawen) [inline pylib/gyp/input.py:983]: In this case, maybe concat stderr info too ?

**Review by gengjiawen — APPROVED** [review]

File: pylib/gyp/input.py (hunk 1 above)
Comment (FuPeiJiang) [reply pylib/gyp/input.py:983]: if it's merged, do I ignore this ?
___
concat stderr to what ?

to `replacement` ?
`replacement = result.stdout.decode("utf-8").rstrip() + "\n" + result.stderr.decode("utf-8").rstrip()`
the order will be changed, and I don't know how to keep original order as it appeared on screen
___
the original code did not concat stderr

Comment (gengjiawen) [reply pylib/gyp/input.py:983]: original code do this in `sys.stderr.write(p_stderr)`

> replacement = result.stdout.decode("utf-8").rstrip() + "\n" + result.stderr.decode("utf-8").rstrip()

Maybe put stderr first.

>if it's merged, do I ignore this ?

you can do a new PR.

Comment (FuPeiJiang) [reply pylib/gyp/input.py:983]: the original code is redundant:
it captures stderr,
`stderr=subprocess.PIPE,`
if stderr is not empty string
`if p.wait() != 0 or p_stderr:`
it writes it back to stderr
`sys.stderr.write(p_stderr)`
(it was only captured for condition checking)

the new code:
just don't capture it, stderr is written to stderr anyways

assuming `replacement`(stdout) is also written to stderr,
if you don't capture stderr, you don't have to concat it to stderr
___
> Maybe put stderr first.

I don't see the point of this concatenation, what does it even change ?

**Pull request merged**